django/db/backends/base/base.py

```diff
@@ -492,6 +708,43 @@ def set_autocommit(
             self.run_and_clear_commit_hooks()
             self.run_commit_hooks_on_set_autocommit_on = False
 
+    async def aset_autocommit(
+        self, autocommit, force_begin_transaction_with_broken_autocommit=False
+    ):
+        """
+        Enable or disable autocommit.
+
+        The usual way to start a transaction is to turn autocommit off.
+        SQLite does not properly start a transaction when disabling
+        autocommit. To avoid this buggy behavior and to actually enter a new
+        transaction, an explicit BEGIN is required. Using
+        force_begin_transaction_with_broken_autocommit=True will issue an
+        explicit BEGIN with SQLite. This option will be ignored for other
+        backends.
+        """
+        self.validate_no_atomic_block()
+        await self.aclose_if_health_check_failed()
+        await self.aensure_connection()
+
+        start_transaction_under_autocommit = (
+            force_begin_transaction_with_broken_autocommit
+            and not autocommit
+            and hasattr(self, "_start_transaction_under_autocommit")
+        )
+
+        if start_transaction_under_autocommit:
+            self._start_transaction_under_autocommit()
```

Comment:
ok, so.. `_start_transaction_under_autocommit()` is a method that's only defined for the SQLite backend, which has no async-support planned (to the best of my knowledge), and therefore `new_connection` would fail above the stack before ever getting here.

Reply:
I mean there is https://github.com/omnilib/aiosqlite so it’s not completely out of the picture for Async support for sqlite

Reply:
Correct. I simply meant that it's not something that has been discussed, _yet_ :)